### PR TITLE
Update position and options

### PR DESCRIPTION
**Pull request by zackbloom**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -47,75 +46,98 @@
...
+        template = strings.oneMinute
+      else
+        template = strings.manyMinutes
     }
 
-    rendering = false
+    if (template){
+      textContainer.innerText = template.replace(/\$MINUTES/g, roundedMinutes);
+    } else {
+      element.style.opacity = 0
```
Comment (zackbloom) [inline app.js]: This allows the user to clear out the text for a given message if they don't wish it to be shown.

File: app.js [hunk 2]
```diff
@@ -47,75 +46,98 @@
...
 
   function updateElement() {
     if (element && element.parentNode) element.parentNode.removeChild(element)
 
-    element = d.createElement("div")
-    element.className = "eager-reading-time"
+    element = d.createElement("eager-app")
+    element.className = `eager-reading-time eager-reading-time-${ options.position }`
+
+    textContainer = d.createElement("div")
```
Comment (zackbloom) [inline app.js]: The inner element is useful for styling.

File: app.js [hunk 3]
```diff
@@ -47,75 +46,98 @@
...
+
+    if (options.showBackground)
+      textContainer.style.backgroundColor = options.backgroundColor;
 
     d.body.appendChild(element)
 
     observer && observer.disconnect()
-    window.removeEventListener("scroll", render, true)
+    window.removeEventListener("scroll", render)
+    window.removeEventListener("resize", render)
```
Comment (zackbloom) [inline app.js]: Resize events also update the scroll position.

File: dist/app.css [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
-  padding: 5px 10px;
   position: fixed;
   opacity: 0;
-  right: 20px;
-  top: 0;
-  transition: opacity 150ms linear;
+  transition: opacity 150ms ease-in-out;
   z-index: 999999;
+  text-align: left;
+  pointer-events: none;
```
Comment (zackbloom) [inline dist/app.css]: Without this the user can't click anything below the message, even when it's opacity is 0.

File: dist/app.css [hunk 5]
```diff
@@ -1,24 +1,40 @@
...
+}
+
+.eager-reading-time-center-top, .eager-reading-time-center-bottom {
+  text-align: center;
+  left: 0;
+  width: 100%;
+}
+
+.eager-reading-time > div {
+  padding: 1em 1.5em;
```
Comment (zackbloom) [inline dist/app.css]: Using `em`s allows the padding to adjust with the font size.

File: app.js [hunk 6]
```diff
@@ -47,75 +46,98 @@
...
-      element.innerText = roundedMinutes > 1 ? roundedMinutes + " minutes left" : "1 minute left"
+      if (roundedMinutes === 1)
+        template = strings.oneMinute
+      else
+        template = strings.manyMinutes
     }
 
-    rendering = false
+    if (template){
+      textContainer.innerText = template.replace(/\$MINUTES/g, roundedMinutes);
```
Comment (TeffenEllis) [inline app.js]: Consider ES6 templating.

File: app.js [hunk 7]
```diff
@@ -47,75 +46,98 @@
...
-    opacityTimeout = setTimeout(() => { element.style.opacity = 0 }, options.visibleDuration)
+    if (options.visibleDuration !== "-1"){
+      opacityTimeout = setTimeout(() => { element.style.opacity = 0 }, +options.visibleDuration)
+    }
 
     const {minutes, wordCount} = getTextEstimates(target.innerText, getScrollPercentage(target))
 
+    let strings = options.strings;
+    if (!strings || !options.localize){
+      strings = {
```
Comment (TeffenEllis) [inline app.js]: Consider the following,

```javascript
const strings = {
  finished: options.strings.finished || "",
  lessThanAMinute: options.strings.lessThanAMinute || "",
  ...
}
```

File: app.js (hunk 3 above)
Comment (TeffenEllis) [inline app.js]: Good catch!

File: app.js [hunk 8]
```diff
@@ -47,75 +46,98 @@
...
 
-    element.style.transform = `translateY(${getScrollBarPosition()}px)`
     element.style.opacity = 1
 
-    opacityTimeout = setTimeout(() => { element.style.opacity = 0 }, options.visibleDuration)
+    if (options.visibleDuration !== "-1"){
+      opacityTimeout = setTimeout(() => { element.style.opacity = 0 }, +options.visibleDuration)
+    }
+
+    const {minutes, wordCount} = getTextEstimates(target.textContent, getScrollPercentage(target))
```
Comment (zackbloom) [inline app.js]: `innerText` is not supported in most versions of Firefox.

File: dist/app.css (hunk 4 above)
Comment (TeffenEllis) [inline dist/app.css]: Nitpick: Alphabetize properties.

File: app.js [hunk 9]
```diff
@@ -47,75 +46,98 @@
...
+    }
   }
 
   function updateElement() {
     if (element && element.parentNode) element.parentNode.removeChild(element)
 
-    element = d.createElement("div")
-    element.className = "eager-reading-time"
+    element = d.createElement("eager-app")
+    element.className = `eager-reading-time eager-reading-time-${ options.position }`
```
Comment (TeffenEllis) [inline app.js]: Consider the following,

```javascript
element.className = "eager-reading-time"
element.setAttribute("data-position", options.position)
```

File: app.js (hunk 6 above)
Comment (zackbloom) [inline app.js]: I don't see a way to do that which wouldn't be a big security issue.  It also would require evaling the template string expression, no?

File: app.js (hunk 7 above)
Comment (zackbloom) [inline app.js]: The goal is to allow the user to specify an empty string for messages they don't want to display.

File: app.js (hunk 9 above)
Comment (zackbloom) [inline app.js]: Doing this ended up making the css a lot better b/c I could use the `~=` comparison operator, thanks!

**Pull request merged**